src/goto-analyzer/show_on_source.cpp

```diff
@@ -0,0 +1,156 @@
+/*******************************************************************\
+
+Module: goto-analyzer
+
+Author: Daniel Kroening, kroening@kroening.com
+
+\*******************************************************************/
+
+#include "show_on_source.h"
+
+#include <util/file_util.h>
+#include <util/message.h>
+#include <util/unicode.h>
+
+#include <analyses/ai.h>
+
+#include <fstream>
+
+/// get a path to the file in given source location
+static std::string file(const source_locationt &l)
```

Comment:
I think this utility function should be moved to `util/source_location.{cpp,h}` as 1) it may be useful in other places and 2) it builds on information that is specific to `source_locationt`, and not to what's happening in this current file.